contracts/DecentHats.sol

```diff
@@ -0,0 +1,485 @@
+            _createSablierStream(hat.sablierParams[i], accountAddress);
+            unchecked {
+                ++i;
+            }
+        }
+    }
+
+    function _createTermedHatAndAccountAndMintAndStreams(
+        IHats hatsProtocol,
+        address topHatAccount,
+        address eligibilityAddress,
+        uint256 adminHatId,
+        Hat calldata hat
+    ) internal {
+        require(
+            hat.termedParams.length == 1,
+            "DecentHats: termedParams length must be 1"
+        );
+        require(
+            hat.termedParams[0].nominatedWearers.length == 1,
+            "DecentHats: nominatedWearers length must be 1"
+        );
+        uint256 hatId = _createHat(
+            hatsProtocol,
+            adminHatId,
+            hat,
+            topHatAccount,
+            eligibilityAddress
+        );
+
+        IHatsElectionEligibility(eligibilityAddress).elect(
+            hat.termedParams[0].termEndDateTs,
+            hat.termedParams[0].nominatedWearers
+        );
+
+        hatsProtocol.mintHat(hatId, hat.termedParams[0].nominatedWearers[0]);
+
+        for (uint256 i = 0; i < hat.sablierParams.length; ) {
+            _createSablierStream(
+                hat.sablierParams[i],
+                hat.termedParams[0].nominatedWearers[0]
+            );
+            unchecked {
+                ++i;
+            }
+        }
+    }
+
+    function _createAdminHatAndAccount(
+        IHats hatsProtocol,
+        IERC6551Registry registry,
+        ModuleProxyFactory moduleProxyFactory,
+        address decentAutonomousAdminMasterCopy,
+        address hatsAccountImplementation,
+        address topHatAccount,
+        uint256 topHatId,
+        Hat calldata hat
+    ) internal returns (uint256 adminHatId, address accountAddress) {
+        adminHatId = _createHat(
+            hatsProtocol,
+            topHatId,
+            hat,
+            topHatAccount,
+            topHatAccount
+        );
+
+        accountAddress = _createAccount(
+            registry,
+            hatsAccountImplementation,
+            address(hatsProtocol),
+            adminHatId
+        );
+
+        hatsProtocol.mintHat(
+            adminHatId,
+            moduleProxyFactory.deployModule(
+                decentAutonomousAdminMasterCopy,
+                abi.encodeWithSignature("setUp(bytes)", bytes("")),
+                uint256(keccak256(abi.encodePacked(SALT, adminHatId)))
+            )
+        );
+    }
+
+    function _createElectionEligiblityModule(
+        IHatsModuleFactory hatsModuleFactory,
+        address hatsElectionEligibilityImplementation,
+        uint256 hatId,
+        uint256 topHatId,
+        TermedParams calldata termedParams
+    ) internal returns (address electionModuleAddress) {
+        electionModuleAddress = hatsModuleFactory.createHatsModule(
+            hatsElectionEligibilityImplementation,
+            hatId,
+            abi.encode(topHatId, uint256(0)),
```

Comment:
Does this 0 need to be cast? Also, can you please add some comments on these lines about what these "immutable" and "mutable" args represent, from the PoV of the Elections Eligibility contract?